_posts/2019-01-23-multifile-python-functions.md

```diff
@@ -0,0 +1,213 @@
+    map(str.lower,
+        map(str.strip,
+            open(os.path.join(FILE, 'stopwords'), 'r').readlines()
+            )
+        )
+)
+
+
+def process_text(text: bytes) -> Dict[str, int]:
+    """Splits a long text into words a count of interesting words in the text.
+
+    process_text will eliminate any of the stopwords, punctuation, and normalize
+    the text to merge cases and plurals into a single value.
+    """
+    words = text.decode("utf-8").split()
+    # remove stopwords
+    # remove 's
+    words = [
+        word[:-2] if word.lower().endswith(("'s",))
+        else word
+        for word in words
+    ]
+    # remove numbers
+    words = [word for word in words if not word.isdigit()]
+    words = [strip_punctuation(word) for word in words]
+    words = [word for word in words if word.lower() not in STOPWORDS]
+
+    return process_tokens(words)
+
+def process_tokens(words: List[str]) -> Dict[str, int]:
+    """Normalize cases and remove plurals.
+    """
+    # ...
+
+def strip_punctuation(text: str) -> str:
+    # ...
+```
+
+Note that processing the `STOPWORDS` at the start of the file means it will be loaded into memory once the package is imported. In this case, once OpenFaaS deploys the function.
+
+Using an explicit relative import, we can very easily use the `process_text` method to create a very simple `handler.py`:
+
+```py
+# handler.py
+import json
+from .wordcloud import process_text
+
+
+def handle(req):
+    """handle a request to the function
+    Args:
+        req (str): request body
+    """
+
+    return json.dumps(process_text(req))
+```
+
+This style of relative imports will work for any file or sub-package you include inside of your function folder. Additionally, your IDE and linter will be able to resolve the imported code correctly!
+
+## Setup OpenFaaS
+
+OpenFaaS on Kubeneretes is my go-to deployment because I can use the built in k8s cluster in Docker-for-Mac.
+
+```sh
+$ kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
+namespace "openfaas" created
+namespace "openfaas-fn" created
+
+$ helm repo add openfaas https://openfaas.github.io/faas-netes/ && helm repo update
+"openfaas" has been added to your repositories
+Hang tight while we grab the latest from your chart repositories...
+...Successfully got an update from the "incubator" chart repository
+...Successfully got an update from the "contiamo" chart repository
+...Successfully got an update from the "openfaas" chart repository
+...Successfully got an update from the "stable" chart repository
+Update Complete. ⎈ Happy Helming!⎈
+
+$ helm upgrade openfaas --install openfaas/openfaas \
+    --namespace openfaas  \
+    --set basic_auth=false \
+    --set faasnetesd.imagePullPolicy='IfNotPresent' \
+    --set functionNamespace=openfaas-fn
+```
+
+This creates an OpenFaaS installation with auth disabled and it will use my locally built images for my functions.
+
+## Deploy and test the function
+
+```sh
```

Comment:
Let's use `faas-cli up` and have the user push to the Docker hub, this is not a confidential example and we've documented private registries.

If you'd like to link to using private Docker registries that would make sense.

Reply:
The only issue with using `faas-cli up` is that it assumes the reader already has a docker account and that they have properly configured their local environment. The current walkthrough should not need any user owned docker registries (public or private)

Reply:
They shouldn't edit the stack.yml file, they should use --prefix when doing `faas-cli new`. Using a registry is the core part of the OpenFaaS experience and we can't gloss over it.

Reply:
i have modified it